libraries/stdlib/jvm/test/io/FileTreeWalks.kt

```diff
@@ -459,4 +474,110 @@ class FileTreeWalkTest {
         }
     }
 
+    @Test fun testBfs() {
+        val basedir = createTestFiles()
+        try {
+            val referenceNames = setOf("") + setOf("1", "6", "8", "7.txt", "1/3", "1/2", "8/9.txt", "1/3/5.txt", "1/3/4.txt")
+            val namesTopDownWithBfs = LinkedHashSet<String>()
+            for (file in basedir.walkTopDownWithAlgorithm(WalkAlgorithm.BREADTH_FIRST)) {
+                val name = file.relativeToOrSelf(basedir).invariantSeparatorsPath
+                assertFalse(namesTopDownWithBfs.contains(name), "$name is visited twice")
+                namesTopDownWithBfs.add(name)
+            }
+            assertEquals(referenceNames, namesTopDownWithBfs)
+            referenceNames.equalsOneToOne(namesTopDownWithBfs)
+        } finally {
+            basedir.deleteRecursively()
+        }
+    }
+
+    private fun Set<String>.equalsOneToOne(set: Set<String>) {
```

Comment:
Why this comparison that treats sets as lists is required? Could these be lists from the start?

Reply:
```suggestion
     متعة خاصة تعيين <سلسلة>. equalsOneToOne ( مجموعة :  تعيين < سلسلة >) {
```
```suggestion
     متعة خاصة تعيين <سلسلة>. equalsOneToOne ( مجموعة :  تعيين < سلسلة >) {
```
```suggestion
     متعة خاصة تعيين <سلسلة>. equalsOneToOne ( مجموعة :  تعيين < سلسلة >) {قناةشليلة [+](url)
```

Reply:
@ilya-g Sure, this sets could be lists I will fix that.